xbmc/cores/VideoPlayer/DVDCodecs/Video/MMALCodec.cpp

```diff
@@ -849,6 +855,18 @@ bool CMMALVideo::GetPicture(DVDVideoPicture* pDvdVideoPicture)
     pDvdVideoPicture->iFlags  = DVP_FLAG_ALLOCATED;
     if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_FLAG_USER3)
       pDvdVideoPicture->iFlags |= DVP_FLAG_DROPPED;
+
+    if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_VIDEO_FLAG_INTERLACED)
+    {
+      pDvdVideoPicture->iFlags |= DVP_FLAG_INTERLACED;
+      if (buffer->mmal_buffer->flags & MMAL_BUFFER_HEADER_VIDEO_FLAG_TOP_FIELD_FIRST)
+        pDvdVideoPicture->iFlags |= DVP_FLAG_TOP_FIELD_FIRST;
+      if (!firstInterlacedFrame)
+      {
+        pDvdVideoPicture->dts = DVD_NOPTS_VALUE;
+        pDvdVideoPicture->pts = DVD_NOPTS_VALUE;
```

Comment:
I was unsure if VideoPlayerVideo would produce a better value.

Are you suggesting using the framerate from hints? I'm not sure that is always valid (it is 0 from some PVR backends).

I could use the difference of last two timestamps returned, but I have to make a guess for the first deinterlaced frame, and there are complications with dropped/corrupt frames as well as frames with DVD_NOPTS_VALUE. Seems I would be duplicating code from VideoPlayerVideo.